### PR TITLE
add patch to load rb3enhanced if exists

### PR DESCRIPTION
**Pull request by jnackmclain**
This patch requires a build of Xenia Canary that supports writable_code_segments, and having it enabled in the config.
Commit 28f3eb6 or later of RB3Enhanced recommended, due to removing the DEMO check.

### CODE REVIEW

File: patches/45410914 - Rock Band 3.patch.toml [hunk 1]
```diff
@@ -11,3 +15,25 @@ hash = "88E6DC86853C2522" # default.xex
     [[patch.be32]]
         address = 0x8226045c
         value = 0x485b5ddc
+
+# This patch requires a build of Xenia Canary that supports writable_code_segments, and having it enabled in the config.
+# Commit 28f3eb6 or later of RB3Enhanced recommended, due to removing the DEMO check.
```
Comment (Margen67) [inline patches/45410914 - Rock Band 3.patch.toml:20]: This should be in the description.

File: patches/45410914 - Rock Band 3.patch.toml [hunk 2]
```diff
@@ -11,3 +15,25 @@ hash = "88E6DC86853C2522" # default.xex
...
+		value = 0x3C6082C6 # lis r3, 0x82C64350@hi
+	[[patch.be32]]
+		address = 0x8283CD68
+		value = 0x60634350 # ori r3, r3, 0x82C64350@lo
+	[[patch.be32]]
+		address = 0x8283CD6C
+		value = 0x48007F0D # bl LoadLibrary
+	[[patch.string]]
+		address = 0x82C64350
+		value = "RB3Enhanced.dll"
```
Comment (illusion0001) [inline patches/45410914 - Rock Band 3.patch.toml:39]: EOF and uppercase hex, needs fixing

Comment (jnackmclain) [reply patches/45410914 - Rock Band 3.patch.toml:39]: Apologies, but what does the EOF comment mean in this instance. Thank you

Comment (jnackmclain) [reply patches/45410914 - Rock Band 3.patch.toml]: ah, got it

**Comment by jnackmclain** [comment]
@Margen67 @illusion0001 changes completed

File: patches/45410914 - Rock Band 3.patch.toml (hunk 2 above)
Comment (illusion0001) [reply patches/45410914 - Rock Band 3.patch.toml]: Not fixed

Comment (jnackmclain) [reply patches/45410914 - Rock Band 3.patch.toml]: my bad. now actually fixed

**Review by illusion0001 — APPROVED** [review]

**Comment by InvoxiPlayGames** [comment]
The "skip SEH check" patch is for the base game's default.xex while the RB3Enhanced one is for TU5 - perhaps worth splitting this patch off into a file for TU5 specifically, while leaving the old patch for the base game?

0x82272e90: 0x4bffd1f1 for a TU5 equivalent patch.

**Comment by Margen67** [comment]
> The "skip SEH check" patch is for the base game's default.xex while the RB3Enhanced one is for TU5 - perhaps worth splitting this patch off into a file for TU5 specifically, while leaving the old patch for the base game?
> 
> 0x82272e90: 0x4280d1f1 for a TU5 equivalent patch.

Creating a separate patch for TU5 would be fine.

**Comment by jnackmclain** [comment]
I accidently deleted my repo, mistakenly forgetting this PR was never merged. My apologies. I created a new PR, #243 which keeps the original patch the same for base 1.0 xex, and instead adds a new patch for TU applied xex as requested

**Pull request closed without merging**